haystack/components/tools/tool_invoker.py

```diff
@@ -111,20 +126,35 @@ def __init__(self, tools: List[Tool], raise_on_failure: bool = True, convert_res
         :param convert_result_to_json_string:
             If True, the tool invocation result will be converted to a string using `json.dumps`.
             If False, the tool invocation result will be converted to a string using `str`.
-
         :raises ValueError:
             If no tools are provided or if duplicate tool names are found.
         """
-
         if not tools:
-            raise ValueError("ToolInvoker requires at least one tool to be provided.")
-        _check_duplicate_tool_names(tools)
```

Comment:
I would prefer to using `_check_duplicate_tool_names` unless it is problematic

Reply:
Removing it was not on purpose.